src/libraries/System.Runtime/tests/System/Runtime/Versioning/OSPlatformAttributeTests.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Xunit;
+
+namespace System.Runtime.Versioning.Tests
+{
+    public class OSPlatformAttributeTests
+    {
+        [Theory]
+        [InlineData("Windows10.0")]
+        [InlineData("iOS")]
+        [InlineData("")]
+        public void TestTargetPlatformAttribute(string platformName)
+        {
+            var tpa = new TargetPlatformAttribute(platformName);
+
+            Assert.Equal(platformName, tpa.PlatformName);
+        }
+
+        [Theory]
+        [InlineData("Windows8.0", "Obsolete", "http://test.com/obsoletedInOSPlatform")]
+        [InlineData("Linux", "Message", null)]
+        [InlineData("iOS13", null, null)]
+        [InlineData("", null, "http://test.com/obsoletedInOSPlatform")]
+        public void TestObsoletedInOSPlatformAttribute(string platformName, string message, string url)
+        {
+            var opa = message == null ? new ObsoletedInOSPlatformAttribute(platformName) { Url = url} : new ObsoletedInOSPlatformAttribute(platformName, message) { Url = url };
+
+            Assert.Equal(platformName, opa.PlatformName);
+            Assert.Equal(message, opa.Message);
+            Assert.Equal(url, opa.Url);
+        }
+
+        [Theory]
+        [InlineData("Windows8.0")]
+        [InlineData("Android4.1")]
```

Comment:
@mhutch I'm not sure if we specified this in the spec but are we going to use the OS version for Android like in this example or the API level?

Reply:
It's important that we're consistent across TFM and `[MinimumOSPlatform]`. IOW, if we want to use the API level for `[MinimumOSPlatform]` then we we need to use as the TFM version too. Otherwise the analyzer would have to know which OS version supports which API level, which is a mapping we mostly like don't want.

However, the same applies to `RuntimeInformation.IsOSPlatformXxx` APIs. At runtime we need to be able to tell what API level the OS is at.

Reply:
Yeah and I can't find any spot in the spec where we've made that decision, I've opened https://github.com/dotnet/designs/issues/136 to clarify that.